va/validation-authority.go

```diff
@@ -54,6 +54,26 @@ type verificationRequestEvent struct {
 
 // Validation methods
 
+// parseDNSError checks the error returned from Lookup... methods and tests if the error
+// was an underlying net.OpError or an error caused by resolver returning SERVFAIL or other
+// invalid Rcodes.
+func (va ValidationAuthorityImpl) parseDNSError(err error, challenge core.Challenge) core.Challenge {
```

Comment:
Also, if you pass the challenge by reference, you'll save yourself the overhead of copying it and re-assigning it.